src/main/java/org/webjars/WebJarVersionLocator.java

```diff
@@ -17,70 +18,112 @@ public class WebJarVersionLocator {
     /**
      * The path to where webjar resources live.
      */
-    public static final String WEBJARS_PATH_PREFIX = "META-INF/resources/webjars";
+    public final String WEBJARS_PATH_PREFIX = "META-INF/resources/webjars";
 
-    private static final String PROPERTIES_ROOT = "META-INF/maven/";
-    private static final String NPM = "org.webjars.npm/";
-    private static final String PLAIN = "org.webjars/";
-    private static final String POM_PROPERTIES = "/pom.properties";
+    private final String PROPERTIES_ROOT = "META-INF/maven/";
+    private final String NPM = "org.webjars.npm/";
+    private final String PLAIN = "org.webjars/";
+    private final String POM_PROPERTIES = "/pom.properties";
 
-    private static final ClassLoader LOADER = WebJarVersionLocator.class.getClassLoader();
+    private final ClassLoader LOADER = WebJarVersionLocator.class.getClassLoader();
 
+    private final WebJarCache cache;
 
-    @Nullable
-    public static String fullPath(final String webJarName, final String exactPath) {
-        String version = webJarVersion(webJarName);
-        String fullPath = String.format("%s/%s/%s", WEBJARS_PATH_PREFIX, webJarName, exactPath);
-        if (!isEmpty(version)) {
-            if (!exactPath.startsWith(version)) {
-                fullPath = String.format("%s/%s/%s/%s", WEBJARS_PATH_PREFIX, webJarName, version, exactPath);
-            }
-        }
+    public WebJarVersionLocator() {
+        this.cache = new WebJarCacheDefault(new ConcurrentHashMap<>());
+    }
+
+    public WebJarVersionLocator(WebJarCache cache) {
+        this.cache = cache;
+    }
 
-        if (LOADER.getResource(fullPath) != null) {
-            return fullPath;
+    public static class DEFAULT {
+        private static final WebJarVersionLocator webJarVersionLocator = new WebJarVersionLocator();
```

Comment:
Any reason why DEFAULT is a static class not a field? Just curious. The API for users is the same but I don’t see this idiom anywhere else.

Reply:
Yeah, I think you are right that it could just be a field.  I'll give that a try.

Reply:
Ok, this change is now in.